subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/repositories_cheat_sheet.adoc

```diff
@@ -0,0 +1,425 @@
+
+* `https.proxyHost`
+* `https.proxyPort`
+* `https.proxyUser`
+* `https.proxyPassword`
+* `http.nonProxyHosts` (NOTE: this is not a typo.)
+
+If the `org.gradle.s3.endpoint` property has been specified with a HTTP (not HTTPS) URI the following system proxy settings can be used:
+
+* `http.proxyHost`
+* `http.proxyPort`
+* `http.proxyUser`
+* `http.proxyPassword`
+* `http.nonProxyHosts`
+
+[[sub:s3_v4_signatures]]
+=== AWS S3 V4 Signatures (AWS4-HMAC-SHA256)
+
+Some of the AWS S3 regions (eu-central-1 - Frankfurt) require signing all HTTP requests in accordance with AWS's http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html[signature version 4].
+
+Always specify S3 URL's containing the region specific endpoint when using buckets that require V4 signatures:
+
+----
+s3://somebucket.s3.eu-central-1.amazonaws.com/maven/release
+----
+
+When a region-specific endpoint is _not_ specified for buckets requiring V4 Signatures, Gradle uses the default AWS region (us-east-1) and the
+following warning appears on the console:
+
+----
+Attempting to re-send the request to .... with AWS V4 authentication.
+To avoid this warning in the future, use region-specific endpoint to
+access buckets located in regions that require V4 signing.
+----
+
+Failure to specify the region-specific endpoint for buckets requiring V4 signatures means:
+
+* 3 round-trips to AWS for every file upload and download.
+* Increased network latencies and slower builds.
+* Increased likelihood of transmission failures.
+
+Always specify a region-specific endpoint.
+
+[[sub:s3_cross_account]]
+==== AWS S3 Cross Account Access
+
+Some organizations may have multiple AWS accounts, e.g. one for each team.
+The AWS account of the bucket owner is often different from the artifact publisher and consumers.
+The bucket owner must grant consumers access, otherwise artifacts are only usable from the publisher's account.
+To grant consumers access, add the `bucket-owner-full-control` link:https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl[Canned ACL] to the uploaded objects.
+Gradle will do this in every upload.
+Make sure the publisher has the required IAM permissions either directly or via an assumed IAM Role:
+
+* `PutObjectAcl`
+* `PutObjectVersionAcl` if bucket versioning is enabled
+
+For more informration, see link:https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-access-control.html[AWS S3 Access Permissions].
+
+[[sec:gcs-repositories]]
+== Declare a Google Cloud Storage Repository
+
+Use the following system properties to configure link:https://cloud.google.com/storage/[Google Cloud Storage] (GCS) repositories:
+
+`org.gradle.gcs.endpoint`::
+Used to override the GCS endpoint when using a non-Google Cloud Platform (GCP), GCS API compatible, storage service.
+
+`org.gradle.gcs.servicePath`::
+Used to override the GCS root service path which the Google Cloud Storage client builds requests from, defaults to `/`.
```

Comment:
nitpick:
```suggestion
Used to override the GCS root service path, which the Google Cloud Storage client builds requests from, defaults to `/`.
```

Reply:
```suggestion
Used to override the GCS root service path which the Google Cloud Storage client builds requests from. Defaults to `/`.
```
This sentence is still a bit rambly. Breaking it up into two instead.